File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    s = list(s)
    stack = []
    for i in s:
        if i == '(':
            stack.append('(')
        else:
            try:
                if stack.pop() == ')':
                    return False
            except:
                return False
    return False if stack else True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        int count = 0;
        
        for (char i : s.toCharArray()) {
            if (i == '(') {
                count += 1;
            } else {
                if (count <= 0) return false;
                count -= 1;
            }
        }

        return count == 0 ? true : false;
    }
}